fact/src/bpf/mod.rs

```diff
@@ -90,19 +97,19 @@ impl Bpf {
                         let ringbuf = guard.get_inner_mut();
                         while let Some(event) = ringbuf.next() {
                             let event: &event_t = unsafe { &*(event.as_ptr() as *const _) };
-                            let event: Event = event.try_into().unwrap();
+                            let event: Arc<Event> = Arc::new(event.try_into().unwrap());
```

Comment:
Not strictly related to your changes but I've seen an error here before about the event not parsing properly, and I wonder if for robustness reasons we can attempt to convert the event and log/continue if it fails?


```suggestion
                            let event: Result<Event, anyhow::Error> = event.try_into();
                            let event: Arc<Event> = if let Ok(parsed) = event {
                                Arc::new(event.try_into().unwrap())
                            } else {
                                error!("Failed to parse event: {:?}", event);
                                continue;
                            };
```

Reply:
I'll do something similar, we can use `match Event::try_from(event){}` and that will be more readable IMO.

Reply:
Actually, on second thought, I don't know that printing `event` with debug information is very useful, since it has multiple 4k buffers that will just spit out garbage. I will move that part to a `debug!` message for now to prevent polluting the logs during normal operation.